utils/tests/utils.go

```diff
@@ -102,3 +104,25 @@ func exitOnErr(err error) {
 		os.Exit(1)
 	}
 }
+
+func InitVcsSubmoduleTestDir(t *testing.T, srcPath string) (submodulePath, tmpDir string) {
+	var err error
+	tmpDir, err = fileutils.CreateTempDir()
+	assert.NoError(t, err)
+
+	err = fileutils.CopyDir(srcPath, tmpDir, true, nil)
+	assert.NoError(t, err)
+	if found, err := fileutils.IsDirExists(filepath.Join(tmpDir, "gitdata"), false); found {
+		assert.NoError(t, err)
+		err := fileutils.RenamePath(filepath.Join(tmpDir, "gitdata"), filepath.Join(tmpDir, ".git"))
+		assert.NoError(t, err)
+	}
+	submoduleDst := filepath.Join(tmpDir, "subDir", "subModule")
+	err = fileutils.CopyFile(submoduleDst, filepath.Join(tmpDir, "gitSubmoduleData"))
+	assert.NoError(t, err)
+	err = os.Rename(filepath.Join(submoduleDst, "gitSubmoduleData"), filepath.Join(submoduleDst, ".git"))
```

Comment:
I think @Or-Geva was recently faced with an issue with ``` os.Rename```, but can't remember what it was. I think he ended up not using this API.
Worth asking him about it.

Reply:
You are right @eyalbe4.
@RobiNino, let's use [this](https://github.com/jfrog/jfrog-client-go/blob/master/utils/io/fileutils/files.go#L571) instead of `os.Rename`